### PR TITLE
Add support for Remote Config parameter groups

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add support for RC Parameter Groups
- Add unit tests
- Add integration tests

### CODE REVIEW

File: src/remote-config/remote-config-api-client.ts [hunk 1]
```diff
@@ -118,6 +125,7 @@ export class RemoteConfigApiClient {
         return {
           conditions: resp.data.conditions,
           parameters: resp.data.parameters,
+          parameterGroups: resp.data.parameterGroups,
```
Comment (hiranya911) [inline src/remote-config/remote-config-api-client.ts:128]: Is this always present on the response?

Comment (lahirumaramba) [reply src/remote-config/remote-config-api-client.ts:128]: If `parameterGroups` is not present then we assign it to an empty object. Same for `conditions` (empty array) and `parameters`. Good catch for tests! Added more unit tests to cover these. Thanks!

Comment (hiranya911) [reply src/remote-config/remote-config-api-client.ts:128]: Also run the code coverage tools and see if anything else has slipped through the cracks.

**Comment by lahirumaramba** [comment]
@hiranya911 Thanks! Added more unit tests to cover cases where `conditions`, `parameters`, and `parameterGroups` are not present in the API response.

**Comment by lahirumaramba** [comment]
@egilmorez please review the docs for Remote Config Parameter Groups. Thank you!!

File: src/index.d.ts [hunk 2]
```diff
@@ -834,6 +834,14 @@ declare namespace admin.remoteConfig {
      */
     parameters: { [key: string]: RemoteConfigParameter };
 
+    /**
```
Comment (lahirumaramba) [inline src/index.d.ts:837]: @egilmorez please find the new docs/comments below. Thanks!

File: test/unit/remote-config/remote-config.spec.ts [hunk 3]
```diff
@@ -394,6 +548,51 @@ describe('RemoteConfig', () => {
...
+      return remoteConfig.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+        .then((template) => {
+          expect(template.conditions).deep.equals([]);
+          expect(template.parameters).deep.equals({});
+          // if parameterGroups are not present in the response, we set it to an empty object.
+          expect(template.parameterGroups).deep.equals({});
+        });
+    });
+
+    it('should resolve with conditions:[] when no conditions present in the response', () => {
```
Comment (hiranya911) [inline test/unit/remote-config/remote-config.spec.ts:581]: There's a lot of redundancy in these tests. As a future improvement, think about ways to clean it up a bit.

File: src/index.d.ts [hunk 4]
```diff
@@ -834,6 +834,14 @@ declare namespace admin.remoteConfig {
...
+     * Map of parameter group names to their parameter group objects.
+     * A group's name is mutable but must be unique among groups in the config.
```
Comment (egilmorez) [inline src/index.d.ts:839]: Not sure, but where we say "the config" here, do we mean the Remote Config template?  Or maybe the `RemoteConfig` object, as you mention it in line 891?

Comment (lahirumaramba) [reply src/index.d.ts]: Good catch. It should be `Remote Config template`. Updated in both places. Thanks!

**Pull request merged**